docs/architecture.md

```diff
@@ -0,0 +1,125 @@
+It is used by the [Mapper](#mapper) to map blocks to the deltas that are collected by the [Feeder](#feeder) component.
+
+[Package documentation](https://pkg.go.dev/github.com/awfm9/flow-dps/chain)
+
+### Filesystem Chain
+
+The Filesystem Chain uses the execution node's on-disk key-value store to reconstruct the block sequence.
+
+TODO: Need a license to be able to link the godoc.
+
+### Network Chain
+
+The Network Chain retrieves data directly from access nodes in order to reconstruct the block sequence.
+
+## Feeder
+
+The Feeder component is responsible for streaming in-order trie updates.
+It outputs deltas which are used by the [Mapper](#mapper) component to map the state trie updates to the block information that the [Chain](#chain) component provides.
+
+[Package documentation](https://pkg.go.dev/github.com/awfm9/flow-dps/streamer)
+
+### Filesystem Feeder
+
+The Filesystem Streamer reads trie updates from the LedgerWAL directly.
+
+### Network Feeder
+
+The Network Streamer receives trie updates through its network subscription on the execution node.
+
+## Mapper
+
+The mapper component is at the core of the DPS. It is responsible for mapping incoming state trie updates to blocks.
+In order to do that, it depends on the [Feeder](#feeder) and [Chain](#chain) components to get state trie updates and block information, as well as on the [Store](#store) component for indexing.
+Generally, trie updates come in by chunk, so each block maps from zero to multiple trie updates.
+Once a block is mapped to its respective trie updates, the mapper forwards the information to the indexer.
+
+[Package documentation](https://pkg.go.dev/github.com/awfm9/flow-dps/mapper)
+
+## Store
+
+The Store component is responsible for receiving a set of trie updates for each block and creating the necessary main indexes and auxiliary in the on-disk database.
+These indexes allow efficient retrieval of the state at arbitrary block heights in the state history.
+It also provides random access to the execution state by providing smart access to these indexes.
+It combines writing and retrieving of indexes, so that an efficient caching strategy is possible.
+
+[Package documentation](https://pkg.go.dev/github.com/awfm9/flow-dps/indexer)
+
+### Database Schema
+
+The DPS uses [BadgerDB](https://github.com/dgraph-io/badger) to store datasets of state changes and block information.
+
+#### Block-To-Height Index
+
+In this index, keys map the block ID to the block height, and they are [prefixed with `1`](https://github.com/awfm9/flow-dps/blob/master/model/prefixes.go#L4).
+
+TODO: Replace link to file and line with a link to the godoc that shows those consts.
+
+| **Length** (bytes) | `1`               | `8`        |
+|:-------------------|:------------------|:-----------|
+| **Type**           | byte              | hex hash   |
+| **Description**    | Index type prefix | Block ID   |
+| **Example Value**  | `1`               | `1FD5532A` |
```

Comment:
I don't like upper case hex strings :D

Reply:
They probably don't like you either! :p Will change it!